hogql_parser/parser.cpp

```diff
@@ -851,68 +858,62 @@ class HogQLParseTreeConverter : public HogQLParserBaseVisitor {
       return visitAsPyObject(placeholder_ctx);
     }
 
-    return visit(ctx->selectUnionStmt());
+    return visit(ctx->selectSetStmt());
   }
 
-  VISIT(SelectUnionStmt) {
-    // Using a vector of PyObjects atypically here, because this is a precursor of flattened_queries
-    vector<PyObject*> select_queries;
-    auto select_stmt_with_parens_ctxs = ctx->selectStmtWithParens();
-    select_queries.reserve(select_stmt_with_parens_ctxs.size());
-    for (auto select_stmt_with_parens_ctx : select_stmt_with_parens_ctxs) {
+  VISIT(SelectSetStmt) {
+    PyObject* initial_query = NULL;
+    PyObject* select_query = NULL;
+    PyObject* select_queries = PyList_New(0);
+    if (!select_queries) {
+      throw PyInternalError();
+    }
+    string set_operator = "";
+
+    for (auto child : ctx->children) {
+      if (auto token = dynamic_cast<antlr4::tree::TerminalNode*>(child)) {
+        if (set_operator == "") {
+          set_operator += child->getText();
+        } else {
+          set_operator += " " + child->getText();
+        }
+        continue;
+      }
```

Comment:
This is super flexible, but also a bit hard to grok. Usually in such cases we do a check like this:
```c++
if (ctx->UNION() && ctx->ALL()) {
	set_operator = "UNION ALL" 
}
```
An example is `VISIT(JoinOpInner)`
(Also, this removes the need for manual uppercasing)

Reply:
In short because of a long stackoverflow thread arguing about how slow boost is compared to this function. But it doesn't matter because I removed it!

Thanks for the push - i refactored the grammar to name the subsequent select clause, and this made the parser code much simpler
![image](https://github.com/user-attachments/assets/55b69d84-f684-458f-923f-7dd5edb60ac0)

![image](https://github.com/user-attachments/assets/45565b95-4dad-4be6-aa95-3002a0dedd07)